### PR TITLE
Add `<UsercentricsSmartDataProtector>` snippet component.

### PR DESCRIPTION
**Pull request by 21christiansc**

### CODE REVIEW

File: src/next/components/Snippet/UsercentricsSmartDataProtector.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		reloadOnOptOut(key: string): void;
+	};
+}>;
+
+export const USERCENTRICS_SERVICE_PROVIDER_KEYS = {
+	GOOGLE_MAPS: "S1pcEj_jZX",
+} as const;
+
+type UsercentricsSerivceProviderInternalKeys = keyof typeof USERCENTRICS_SERVICE_PROVIDER_KEYS;
+type UsercentricsSerivceProviderKey = typeof USERCENTRICS_SERVICE_PROVIDER_KEYS[UsercentricsSerivceProviderInternalKeys] | string;
```
Comment (keichinger) [inline src/next/components/Snippet/UsercentricsSmartDataProtector.tsx:19]: We should prefer using `… | (string & {})` to get proper auto-completion and allow unknown service keys as well. See https://twitter.com/mattpocockuk/status/1823380970147369171 for a demonstration on what it does.

File: src/next/components/Snippet/UsercentricsSmartDataProtector.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+type HTMLSelector = string;
+
+type UsercentricsSmartDataProtectorProps = Readonly<{
+	elements?: Record<UsercentricsSerivceProviderKey, HTMLSelector>;
+}>;
+
+export function UsercentricsSmartDataProtector (props: UsercentricsSmartDataProtectorProps): ReactElement | null
+{
+	const keys = props.elements ? Object.keys(props.elements) : [];
```
Comment (keichinger) [inline src/next/components/Snippet/UsercentricsSmartDataProtector.tsx]: IMO we shouldn't include this script if no services configuration is passed. Or does it make sense to still include it for its automatic logic, which may still be useful?

**Review by keichinger — CHANGES_REQUESTED** [review]

File: src/next/components/Snippet/UsercentricsSmartDataProtector.tsx (hunk 1 above)
Comment (21christiansc) [reply src/next/components/Snippet/UsercentricsSmartDataProtector.tsx:19]: With the current definition, autocompletion is actually already possible.

File: src/next/components/Snippet/UsercentricsSmartDataProtector.tsx (hunk 2 above)
Comment (21christiansc) [reply src/next/components/Snippet/UsercentricsSmartDataProtector.tsx]: It makes sense to include this anyway, as a YouTube, for example, can be recognized automatically. With `elements` you only specify special cases that are not in an iframe on the page, e.g. Google Maps in a div. I have added a description to the prop, which also mentions this, as well as `reloadKeys`, where the non-special cases can also be added to `reloadOnOptIn` and `reloadOnOptOut`.

File: src/next/components/Snippet/UsercentricsSmartDataProtector.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	 * If the key is already used in `elements`, it does not need to be added here again.
+	 */
+	reloadKeys?: UsercentricsSerivceProviderKey[],
+}>;
+
+export function UsercentricsSmartDataProtector (props: UsercentricsSmartDataProtectorProps): ReactElement | null
+{
+	const keys = new Set(
+		[
+			...(props.elements ? Object.keys(props.elements) : []),
```
Comment (keichinger) [inline src/next/components/Snippet/UsercentricsSmartDataProtector.tsx]:
```suggestion
			...(Object.keys(props.elements ?? {})),
```

**Review by keichinger — APPROVED** [review]

**Pull request merged**